stan/math/prim/scal/meta/is_vector_like.hpp

```diff
@@ -22,8 +31,7 @@ namespace stan {
  */
 template <typename T>
 struct is_vector_like
-    : bool_constant<stan::is_vector<T>::value || std::is_pointer<T>::value
-                    || is_eigen<T>::value || std::is_array<T>::value> {};
+    : bool_constant<is_detected<T, internal::operator_bracket_t>::value> {};
```

Comment:
No changes, just want to say that this method of checking ```vector_like```-ness is neat. Very nice!

Reply:
Ty!